### ISSUE TITLE
Add ShadowLooper.runUiThreadTasksIncludingDelayedTasks to RobolectricTest.advanceRobolectricLooper?

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#8643]
if you remove the line and it works, I would commit the removal of the line yes.
Sorry I was unclear, what I was trying (badly) to communicate was that this specific wait method looks different than our current style:

https://github.com/ankidroid/Anki-Android/blob/143d7bbf7c9aa87f842d44c240c07b7cb2bd9285/AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.java#L229-L245

...and as a not-really-related item maybe we should add `ShadowLooper.runUiThreadTasksIncludingDelayedTasks()` to the RobolectricTest advance method(s).

I will peel this comment off to a separate issue for investigation and consider this resolved once you commit the removal of the line in this PR

Thanks @kuroni

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/pull/8579#r615429745_

**Comment by mikehardy** [comment]
(this came from a PR review, and I am not sure it is the right path - but the intention of RobolectricTest.advanceRobolectricLooper is to "wait on all the things so we know all the things are done", so it seems like it should be a good idea?

**Comment by david-allison-1** [comment]
Calls through to `getShadowMainLooper().runToEndOfTasks();` - I don't think this will add value

**Comment by mikehardy** [comment]
resolved -> works for me, thanks for checking, and related PR is unblocked so all clear I think
